src/services/scriptingservice.cpp

```diff
@@ -1986,6 +1995,43 @@ bool ScriptingService::writeToFile(const QString &filePath,
     return true;
 }
 
+/**
+ * Read text from a file
+ *
+ * @param filePath
+ * @return the file data or null if the file does not exist
+ */
+QString ScriptingService::readFromFile(const QString &filePath) const {
+    if (filePath.isEmpty()){
+        return Q_NULLPTR;
```

Comment:
That may crash the application, best return an empty string if you don't want to think of a better way of error handling that works in script.

Reply:
Why, is there no null/undefined in QML?

Reply:
Q_NULLPTR is not a QString. According to your declaration your method needs to return a QString.

Reply:
I don't know of an easy way to return an "undefined" state without pointers.

Reply:
So just an empty QString is NULL?
```
QString().isNull();               // returns true
```

Reply:
You can do that.

Reply:
`QString() // null`
`"" // not null`